File: gin-vue-admin-main/server/router/docker/docker_registry.go
package docker

import (
	api "github.com/flipped-aurora/gin-vue-admin/server/api/v1/docker"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type DockerRegistryRouter struct{}

// InitDockerRegistryRouter 初始化Docker仓库路由
func (d *DockerRegistryRouter) InitDockerRegistryRouter(Router *gin.RouterGroup) {
	dockerRegistryApi := api.DockerRegistryApi{}
	
	// 带操作记录的路由组 - 用于需要记录操作日志的API
	registryRouter := Router.Group("docker").Use(middleware.OperationRecord())
	// 不带操作记录的路由组 - 用于查询类API
	registryRouterWithoutRecord := Router.Group("docker")

	// 需要记录操作的路由（仓库操作）
	{
		registryRouter.POST("registries", dockerRegistryApi.CreateRegistry)                    // 创建仓库
		registryRouter.PUT("registries", dockerRegistryApi.UpdateRegistry)                    // 更新仓库
		registryRouter.DELETE("registries/:id", dockerRegistryApi.DeleteRegistry)             // 删除仓库
		registryRouter.POST("registries/:id/test", dockerRegistryApi.TestRegistry)            // 测试仓库连接
		registryRouter.POST("registries/:id/default", dockerRegistryApi.SetDefaultRegistry)   // 设置默认仓库
		registryRouter.POST("registries/sync", dockerRegistryApi.SyncFrom1Panel)              // 从1Panel同步仓库数据
	}

	// 不需要记录操作的路由（查询类）
	{
		registryRouterWithoutRecord.GET("registries", dockerRegistryApi.GetRegistryList)       // 获取仓库列表
		registryRouterWithoutRecord.GET("registries/:id", dockerRegistryApi.GetRegistryDetail) // 获取仓库详情
	}
}

File: gin-vue-admin-main/web/src/api/dockerRegistry.js
import service from '@/utils/request'

// 获取仓库列表
export const getRegistryList = (data) => {
  return service({
    url: '/docker/registries',
    method: 'get',
    params: data
  })
}

// 获取仓库详情
export const getRegistryDetail = (id) => {
  return service({
    url: `/docker/registries/${id}`,
    method: 'get'
  })
}

// 创建仓库
export const createRegistry = (data) => {
  return service({
    url: '/docker/registries',
    method: 'post',
    data
  })
}

// 更新仓库
export const updateRegistry = (data) => {
  return service({
    url: '/docker/registries',
    method: 'put',
    data
  })
}

// 删除仓库
export const deleteRegistry = (id) => {
  return service({
    url: `/docker/registries/${id}`,
    method: 'delete'
  })
}

// 测试仓库连接
export const testRegistry = (id) => {
  return service({
    url: `/docker/registries/${id}/test`,
    method: 'post'
  })
}

// 设置默认仓库
export const setDefaultRegistry = (id) => {
  return service({
    url: `/docker/registries/${id}/default`,
    method: 'post'
  })
}

